nebula-logger/core/main/logger-engine/lwc/logger/logger.js

```diff
@@ -122,58 +107,37 @@ export default class Logger extends LightningElement {
      */
     @api
     getBufferSize() {
-        return this.componentLogEntries.length;
+        return this.#loggerService.getBufferSize();
     }
 
     /**
      * @description Discards any entries that have been generated but not yet saved
+     * @return {Promise<void>} A promise to clear the entries
      */
     @api
     flushBuffer() {
-        this.componentLogEntries = [];
+        return this.#loggerService.flushBuffer();
     }
 
     /**
-     * @description Saves any entries in Logger's buffer, using the specified save method for only this call.
-     *              All subsequent calls to saveLog() will use the transaction save method.
-     * @param  {String} saveMethod The enum value of Logger.SaveMethod to use for this specific save action.
+     * @description Saves any entries in Logger's buffer, using the specified save method for only this call
+     *              All subsequent calls to saveLog() will use the transaction save method
+     * @param  {String} saveMethod The enum value of Logger.SaveMethod to use for this specific save action
      */
     @api
     saveLog(saveMethodName) {
-        if (this.getBufferSize() > 0) {
-            if (!saveMethodName && this.settings && this.settings.defaultSaveMethodName) {
-                saveMethodName = this.settings.defaultSaveMethodName;
-            }
-
-            saveComponentLogEntries({ componentLogEntries: this.componentLogEntries, saveMethodName: saveMethodName })
-                .then(this.flushBuffer())
-                .catch(error => {
-                    if (this.settings.isConsoleLoggingEnabled === true) {
-                        /* eslint-disable-next-line no-console */
-                        console.error(error);
-                        /* eslint-disable-next-line no-console */
-                        console.error(this.componentLogEntries);
-                    }
-                });
-        }
-    }
-
-    // Private functions
-    _meetsUserLoggingLevel(logEntryLoggingLevel) {
-        let logEntryLoggingLevelOrdinal = this.settings.supportedLoggingLevels[logEntryLoggingLevel];
-        return this.settings && this.settings.isEnabled === true && this.settings.userLoggingLevel.ordinal <= logEntryLoggingLevelOrdinal;
-    }
-
-    _newEntry(loggingLevel, message) {
-        const shouldSave = this._meetsUserLoggingLevel(loggingLevel);
-        const logEntryBuilder = newLogEntry(loggingLevel, shouldSave, this.settings.isConsoleLoggingEnabled).setMessage(message);
-        if (this._scenario) {
-            logEntryBuilder.scenario = this._scenario;
-        }
-        if (this._meetsUserLoggingLevel(loggingLevel) === true) {
-            this.componentLogEntries.push(logEntryBuilder);
-        }
-
-        return logEntryBuilder;
+        this.#loggerService.saveLog(saveMethodName);
     }
 }
+
+/**
+ * @return {LoggerService} a LoggerService instance
+ */
+const getLogger = function () {
```

Comment:
I suggest createLogger to keep consistent with the `createLoggerService` because it clarifies the intent with some work involved other than just fetching a reference and/or pointer.

```suggestion
const createLogger = function () {
```

Reply:
My concern with using `create` is that it's too specific - other components should know or care how the logger instance is being returned. I think that `get` is a more generic verb that helps (purposefully) abstract away the implementation, and would still be accurate if there are ever future enhancements where it _would_ make sense to return a reference/pointer to a singleton instance of logger.  But `createLogger` wouldn't be accurate anymore if (in future releases), references/caching/singleton are ever used.

Reply:
Look, fair enough, sometimes I get into semantics because I feel they matter but let's not get hung up on that and move forward.

Reply:
@jefersonchaves I was out on vacation & didn't have a chance to merge this PR yet - after reviewing it again,I think you're right, using `createLogger()` does help clarify the intent, so I'm reverting from `getLogger()` back to `createLogger()`.